ch11.md

```diff
@@ -0,0 +1,225 @@
+
+What a frightful mess. A collage of abstract types, amateur type expressionism, polymorphic Pollock, monolithic Mondrian. There are many solutions to this common issue. We can compose the types into one monstrous container, sort and `join` a few, homogenize them, deconstruct them, and so on. In this chapter, we'll focus on homogenizing them via *natural transformations*.
+
+## All natural
+
+A *Natural Transformation* is a "morphism between functors", that is, a function which operates on the containers themselves. Typewise, it is a function `(Functor f, Functor g) => f a -> g a`. What makes it special is that we cannot, for any reason, peek at the contents of our functor. Think of it as an exchange of highly classified information - the two parties oblivious to what's in the sealed manila envelope stamped "top secret". This is a structural operation. A functorial costume change. Formally, a *natural transformation* is any function for which the following holds:
+
+<img width=600 src="images/natural_transformation.png" alt="natural transformation diagram" />
+
+or in code:
+
+```js
+// nt :: (Functor f, Functor g) => f a -> g a
+compose(map(f), nt) == compose(nt, map(f))
+```
+
+Both the diagram and the code say the same thing: We can run our natural transformation then `map` or `map` then run our natural transformation and get the same result. Incidentally, that follows from a [free theorem](ch7.md#free-as-in-theorem) though natural transformations (and functors) are not limited to functions on types.
+
+## Principled Type Conversions
+
+As programmers we are familiar with type conversions. We transform types like `Strings` into `Booleans` and `Integers` into `Floats` (though JavaScript only has `Numbers`). The difference here is simply that we're working with algebraic containers and we have some theory at our disposal.
+
+Let's look at some of these as examples:
+
+```js
+// either :: (a -> c) -> (b -> c) -> Either a b -> c
+
+// idToMaybe :: Identity a -> Maybe a
+const idToMaybe = x => Maybe.of(x.__value)
+
+// idToIO :: Identity a -> IO a
+const idToIO = x => IO.of(x.__value)
+
+// eitherToTask :: Either a b -> Task a b
+const eitherToTask = either(Task.rejected, Task.of)
+
+// ioToTask :: IO a -> Task () a
+const ioToTask = x => new Task((reject, resolve) => resolve(x.unsafePerform()))
+
+// maybeToTask :: Maybe a -> Task () a
+const maybeToTask = x => x.isNothing() ? Task.rejected() : Task.of(x.__value)
+
+// arrayToMaybe :: [a] -> Maybe a
+const arrayToMaybe = x => Maybe.of(x[0])
+```
+
+See the idea? We're just changing one functor to another. We are permitted to lose information along the way so long as the value we'll `map` doesn't get lost in the shape shift shuffle. That is the thing here, remember, `map` must carry on after the transformation according to our defining law.
```

Comment:
> That is the thing here, remember, `map` must carry on after the transformation according to our defining law.

Entirely possible that this is just because i'm not a native speaker but i have read the sentence several times now and I still find it confusing -  I know what you are saying but the sentence structure is weird to me ;-)

Reply:
Maybe some comas will help, also by moving ahead the end of the sentence. Something like:

> That is the thing here, remember, map must carry on, according to our definition, even after the transformation.

Reply:
i dig that!